interop-testing/src/main/java/io/grpc/testing/integration/TrafficControlProxy.java

```diff
@@ -0,0 +1,248 @@
+            System.out.println(clientQSize + readableRequestBytes);
+            Thread.sleep(1);
+          }
+          clientQueue.add(m);
+          incrementClientQSize(readableRequestBytes);
+        } catch (IOException e) {
+          e.printStackTrace();
+        } catch (InterruptedException e) {
+          e.printStackTrace();
+        }
+      }
+
+    }
+
+  }
+
+  private class ServerReader implements Runnable {
+
+    DataInputStream serverIn;
+
+    public ServerReader(DataInputStream serverIn) {
+      this.serverIn = serverIn;
+    }
+
+    @Override
+    public void run() {
+      while (true) {
+        try {
+          byte[] response = new byte[chunkSize];
+          int readableResponseBytes = serverIn.read(response);
+          long arrivalTime = System.currentTimeMillis();
+          Message m = new Message(arrivalTime, response, readableResponseBytes);
+          while (serverQSize + readableResponseBytes > qLength) {
+            Thread.sleep(1);
+          }
+          System.out.println("Queueing a server message");
+          serverQueue.add(m);
+          incrementServerQSize(readableResponseBytes);
+          System.out.println("Server q size: " + serverQSize);
+        } catch (IOException e) {
+          e.printStackTrace();
+        } catch (InterruptedException e) {
+          e.printStackTrace();
+        }
+      }
+    }
+  }
+
+  private class ClientWriter implements Runnable {
+
+    DataOutputStream clientOut;
+
+    public ClientWriter(DataOutputStream clientOut) {
+      this.clientOut = clientOut;
+    }
+
+    @Override
+    public void run() {
+      while (true) {
+        try {
+          while (clientQueue.isEmpty()) {
+            Thread.sleep(1);
+          }
+          Message next = clientQueue.get(0);
+          Long howLong = (next.arrivalTime + latency) - System.currentTimeMillis();
+          while (howLong > 0) {
+            Thread.sleep(howLong);
+            howLong = (next.arrivalTime + latency) - System.currentTimeMillis();
+          }
+          clientQueue.remove(0);
+          clientOut.write(next.message, 0, next.messageLength);
+          incrementClientQSize(-next.messageLength);
+        } catch (IOException e) {
+          e.printStackTrace();
+        } catch (InterruptedException e) {
+          e.printStackTrace();
+        }
+      }
+    }
+  }
+
+  private class ServerWriter implements Runnable {
+
+    DataOutputStream serverOut;
+
+    public ServerWriter(DataOutputStream serverOut) {
+      this.serverOut = serverOut;
+    }
+
+    @Override
+    public void run() {
+      while (true) {
+        try {
+          while (serverQueue.isEmpty()) {
+            Thread.sleep(1);
+          }
+          Message next = serverQueue.get(0);
+          Long howLong = (next.arrivalTime + latency) - System.currentTimeMillis();
+          while (howLong > 0) {
+            System.out.println("how long: " + howLong);
+            Thread.sleep(howLong);
```

Comment:
Can you use the sleep overload that accepts nanos?  Or possibly use TimeUnit.sleep?